grafana-plugin/src/containers/ColumnsSelector/ColumnsSelector.tsx

```diff
@@ -0,0 +1,273 @@
+  column: AGColumn;
+  onItemChange: (id: number | string) => void;
+  onColumnRemoval: (column: AGColumn) => void;
+}
+
+const ColumnRow: React.FC<ColumnRowProps> = ({ column, onItemChange, onColumnRemoval }) => {
+  const dnd = useSortable({ id: column.id });
+
+  const styles = useStyles2(getColumnsSelectorStyles);
+
+  const { attributes, listeners, setNodeRef, transform, transition } = dnd;
+  const columnElRef = useRef<HTMLDivElement>(undefined);
+
+  const style: React.CSSProperties = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+  };
+
+  return (
+    <div ref={setNodeRef} style={{ ...style }} className={styles.columnRow}>
+      <div className={styles.columnItem} ref={columnElRef}>
+        <span className={styles.columnName}>{column.name}</span>
+
+        {column.type === AGColumnType.LABEL && (
+          <Tooltip content="Label Column">
+            <Icon aria-label="Label" name="tag-alt" className={styles.labelIcon} />
+          </Tooltip>
+        )}
+
+        {column.isVisible ? (
+          <IconButton
+            aria-label="Drag"
+            name="draggabledots"
+            className={styles.columnsIcon}
+            {...attributes}
+            {...listeners}
+          />
+        ) : column.type === AGColumnType.LABEL ? (
+          <WithPermissionControlTooltip userAction={UserActions.OtherSettingsWrite}>
+            <IconButton
+              className={[styles.columnsIcon, styles.columnsIconTrash, 'columns-icon-trash'].join(' ')}
+              name="trash-alt"
+              aria-label="Remove"
+              tooltip={'Remove column'}
+              onClick={() => onColumnRemoval(column)}
+            />
+          </WithPermissionControlTooltip>
+        ) : undefined}
+      </div>
+
+      <Checkbox
+        className={styles.columnsCheckbox}
+        type="checkbox"
+        value={column.isVisible}
+        onChange={() => onItemChange(column.id)}
+      />
+    </div>
+  );
+};
+
+interface ColumnsSelectorProps {
+  onColumnAddModalOpen(): void;
+  onConfirmRemovalModalOpen(column: AGColumn): void;
+}
+
+export const ColumnsSelector: React.FC<ColumnsSelectorProps> = observer(
+  ({ onColumnAddModalOpen, onConfirmRemovalModalOpen }) => {
+    const { alertGroupStore } = useStore();
+
+    const styles = useStyles2(getColumnsSelectorStyles);
+
+    const { columns } = alertGroupStore;
+
+    const visibleColumns = columns.filter((col) => col.isVisible);
+    const hiddenColumns = columns
+      .filter((col) => !col.isVisible)
+      .sort((a, b) => a.id.toString().localeCompare(b.id.toString()));
+
+    const canResetData = useMemo(() => !isEqual(columns, getDefaultData()), [alertGroupStore.columns]);
+
+    const sensors = useSensors(
+      useSensor(PointerSensor),
+      useSensor(KeyboardSensor, {
+        coordinateGetter: sortableKeyboardCoordinates,
+      })
+    );
+
+    const isResetLoading = LoaderStore.isLoading(ActionKey.IS_RESETING_COLUMNS_FROM_ALERT_GROUP);
```

Comment:
We shouldn't consume the store directly by importing the class but rather through `useStore`. We can also create a custom hook to simplify that. Take a look here: https://github.com/brojd/todo-app/blob/main/frontend/src/store/store.hooks.ts